sdk/python/kfp/_credentials.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Arrikto Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+
+__all__ = [
+    "TokenCredentials",
+]
+
+
+class TokenCredentials(object):
```

Comment:
nit: What do you think about naming it `TokenCredentialsProvider` or `TokenCredentialsBase` or `TokenCredentialsMeta`?

Reply:
Providing more options, WDUT about AuthProvider?

Reply:
From the first suggestions `TokenCredentialsBase` sounds good.

`AuthProvider` is more of a whole backend service that **performs** authentication, rather than a client-side component that ensures the requests contain the required credentials to authenticate against the provider.

Let's go with `TokenCredentialsBase`, but ofc feel free to elaborate!

Reply:
I don't have strong opinions, `TokenCredentialsBase` SGTM